### DISCUSSION TITLE
Is there a possibility to use something else than Imaginary to generate preview for HEIC files ?

### DISCUSSION THREAD

**Discussion by AiroPi** [nextcloud/all-in-one#4510] [Questions] [answered]
It seems than Imaginary has some "bugs" when it generate previews for HEIC files (they are sometimes wrongly oriented for no reasons)
Here is a link to a more complete discussion about the topic : https://github.com/pulsejet/memories/issues/1120

It seems that Nextcloud (and the Preview Generator app) should be able to use another tool to generate previews for HEIC file, which works better than Imaginary.
Could this be possible with Nextcloud AIO ? Why a Nextcloud AIO instance can't use the "default preview pipeline" ?

**Comment by szaimen** [comment, marked as answer]
See https://github.com/nextcloud/all-in-one/pull/4513

**Reply by AiroPi** [reply]
Ok, another question, can we chose Imagick in priority when Imaginary is also enabled ? I don't think AIO has any control over this, and that it would require a change in the app preview itself...
I want to use imaginary for PDFs and Illustrator files, and Imagick for photos.

**Reply by szaimen** [reply]
This would be a feature request for https://github.com/nextcloud/server/issues
